test/AllInOne.cs

```diff
@@ -834,6 +834,14 @@ async void Test()
             }
         }
     }
+
+    class CSharp7Features
+    {
+        int a;
+        // variable assignment is valid inside a tuple literal.
+        var foo = (2, a=1);
```

Comment:
I'd prefer you move this test case among the other tuple-related cases around line 345. No need for a separate class.